includes/Core/User_Input/User_Input_Settings.php

```diff
@@ -0,0 +1,313 @@
+<?php
+/**
+ * Class Google\Site_Kit\Core\User_Input\User_Input_Settings
+ *
+ * @package   Google\Site_Kit\Core\User_Input
+ * @copyright 2022 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\User_Input;
+
+use Google\Site_Kit\Context;
+use Google\Site_Kit\Core\Authentication\Authentication;
+use Google\Site_Kit\Core\Authentication\User_Input_State;
+use Google\Site_Kit\Core\Storage\Options;
+use Google\Site_Kit\Core\Storage\User_Options;
+use Google\Site_Kit\Core\User_Input\User_Input_Site_Settings;
+use Google\Site_Kit\Core\User_Input\User_Input_User_Settings;
+
+/**
+ * Class managing requests to user input settings endpoint.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+class User_Input_Settings {
+
+	/**
+	 * Authentication instance.
+	 *
+	 * @since n.e.x.t
+	 * @var Authentication
+	 */
+	private $authentication;
+
+	/**
+	 * User_Input_Site_Settings instance.
+	 *
+	 * @since n.e.x.t
+	 * @var User_Input_Site_Settings
+	 */
+	protected $user_input_site_settings;
+
+	/**
+	 * User_Input_User_Settings instance.
+	 *
+	 * @since n.e.x.t
+	 * @var User_Input_User_Settings
+	 */
+	protected $user_input_user_settings;
+
+	/**
+	 * User Input properties.
+	 *
+	 * @since n.e.x.t
+	 * @var array|ArrayAccess
+	 */
+	private static $properties = array(
+		'purpose'       => array(
+			'scope' => 'site',
+		),
+		'postFrequency' => array(
+			'scope' => 'user',
+		),
+		'goals'         => array(
+			'scope' => 'user',
+		),
+	);
+
+	/**
+	 * Constructor.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Context        $context         Plugin context.
+	 * @param Authentication $authentication  Optional. Authentication instance. Default a new instance.
+	 */
+	public function __construct( Context $context, Authentication $authentication = null ) {
+		$options      = new Options( $context );
+		$user_options = new User_Options( $context );
```

Comment:
Let's pass these to the constructor where `User_Input_Settings` is instantiated instead of instantiating them here. I think we can do that in the `Authentication` class where we're registering `User_Input`.

Reply:
Done, thank you!